### PR TITLE
Escape SQL LIKE

### LINKED ISSUE #643: rgw/sfs: Prefix to SQL LIKE expression sanitizer/converter
### What needs to be done

We have a couple of places where we need to convert a string prefix into a LIKE expression. Since like uses '%' and '?' as wildcards we need to escape. 

### Acceptance Criteria

1. Utility function
2. Integrated into bucket listing (sqlite_list.cc)
3. Integrated into multipart code

### PR DESCRIPTION
**Pull request by irq0**
Add utility to create LIKE ESCAPE expressions from prefixes. Use that
utility in our list queries.

Fixes: https://github.com/aquarist-labs/s3gw/issues/643

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [x] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [ ] No tests

### CODE REVIEW

File: src/rgw/driver/sfs/sqlite/conversion_utils.h
Comment (0xavi0) [inline src/rgw/driver/sfs/sqlite/conversion_utils.h:133]: Why 10? 
Not important to not approve the PR, but if you want to reserve in advance to avoid future allocations....maybe better to reserve the worst case scenario? (prefix.length() * 2)

10 sounds like a random magic number here. (Or is there a limitation on the length of the prefix? in that case I would name the number)

**Review by 0xavi0 — APPROVED** [review]

File: src/rgw/driver/sfs/sqlite/conversion_utils.h
Comment (irq0) [reply src/rgw/driver/sfs/sqlite/conversion_utils.h:133]: 10 is an educated guess on how many substitutions in common object names. Something to change if it ever turns up in profiling that we reallocate a lot here.

**Pull request merged**
